Gems/Atom/RHI/Vulkan/Code/Source/RHI/Device.cpp

```diff
@@ -815,12 +815,15 @@ namespace AZ
             createInfo.size = descriptor.m_byteCount;
             createInfo.usage = GetBufferUsageFlagBitsUnderRestrictions(descriptor.m_bindFlags);
             // Trying to guess here if the buffers are going to be used as attachments. Maybe it would be better to add an explicit flag in the descriptor.
-            createInfo.sharingMode =
-                RHI::CheckBitsAny(
-                    descriptor.m_bindFlags,
-                    RHI::BufferBindFlags::ShaderWrite | RHI::BufferBindFlags::Predication | RHI::BufferBindFlags::Indirect)
-                ? VK_SHARING_MODE_EXCLUSIVE
-                : VK_SHARING_MODE_CONCURRENT;
+
+            createInfo.sharingMode = 
+                (RHI::CheckBitsAny(
+                    descriptor.m_bindFlags, 
+                    RHI::BufferBindFlags::ShaderWrite | RHI::BufferBindFlags::Predication | RHI::BufferBindFlags::Indirect) || 
+                    static_cast<uint32_t>(queueFamilies.size()) <= 1) ? 
```

Comment:
Question: are there no situations in which we create multiple queues within the same family?

Reply:
Also, why is the static cast needed here?

Reply:
In android case, there is only one queue for the family whereas pc can have multiples.